### ISSUE TITLE
PWAs close first time opening PWA - relative URL without a base

### ISSUE THREAD

**Issue by spoorun** [filips123/PWAsForFirefox#572]
### Description

During any given session, I open a PWA and it momentarily opens than closes
If I open it a second time (from the menu or app icon but not the command line) it opens correctly
This happens whether the PWA is opened from the command line, the KDE menu, or clicking an icon

When opened from the command line, the following error is given:

`/usr/bin/firefoxpwa site launch 01J96HRTE3XC6AW2AQBEFMI7MNMP --protocol %u                                                                                                      error: invalid value '%u' for '--protocol [<PROTOCOL>]': relative URL without a base
For more information, try '--help'.
`
If I remove "%u" from the .desktop file the issue does not occur and the PWA opens correctly first time

### Steps to Reproduce

1. Create a PWA
2. Restart the system (the error only occurs after a restart not when the PWA has just been installed)
3. Launch the PWA from the menu or icon
[It flashes up briefly then fails]
4. Launch again
[It launches correctly]

OR

1. Create a PWA
2. Restart the system (the error only occurs after a restart not when the PWA has just been installed)
3. Launch the PWA from the command line
[error is shown:[ (error: invalid value '%u' for '--protocol [<PROTOCOL>]': relative URL without a base) ]

### Workaround

Remove "%U" from the .desktop file for the PWA
The app starts correctly from menu, icon or commandline

### Affected Websites

All

### Environment

First occurred at least 6 months ago 

* Operating system: Manjaro Linux
* System architecture: 64
* Desktop environment: KDE 6.0 - 6.1
* Installation method: AUR
* PWAsForFirefox extension version: 2.11 - 2.12.4
* PWAsForFirefox native version: 2.11 - 2.12.5
* PWAsForFirefox runtime version: 130.0.1 and earlier
* Firefox version: 129 - 131

**Comment by filips123** [comment]
That may be a KDE bug. `%u` should be replaced with the URL when launching as a protocol handler, or be empty when launching normally. But I've tried to reproduce it on Manjaro with KDE in a VM and it works fine for me...

**Comment by HappyHippo77** [comment]
Happens to me with GNOME on Fedora 41 and on Arch Linux. Seems to be a Linux issue?

**Comment by filips123** [comment]
Unfortunately, I still can't reproduce it neither on GNOME nor KDE. You can try to launch the desktop entry with `gtk-launch FFPWA-ID.desktop` and see if there are any errors.

**Comment by realDayaa** [comment]
Launching with `gtk-launch FFPWA-ID.desktop`  opens the app without errors.

**Comment by probablyanewt** [comment]
I too have this issue. Manjaro and GNOME. Removing %u from the .desktop doesn't resolve it for me though.

**Comment by filips123** [comment]
I guess this might be another session-related problem, but I don't konow why it happens and how to fix it.

**Comment by probablyanewt** [comment]
@filips123 is there a verbose or debug mode I can run it with? I can see if I can get some more info.

**Comment by filips123** [comment]
You can try to get both the [native and runtime logs](https://pwasforfirefox.filips.si/help/troubleshooting/#obtaining-runtime-logs).

**Comment by probablyanewt** [comment]
So in more recent updates, the crash occurs less often than it did previously, which makes it harder to poke.

I added the empty DEBUG file in `~/.local/share/firefoxpwa`, and so far found the following:
- Running `firefoxpwa site launch <id>` didn't write anything in the native log files logs.
- I couldn't get the PWAs to crash when launching with the `-- --jsconsole` flag. I have a couple of dumps from the jsconsole, but I'm not sure how useful they are.
- I did managed to capture an error in the terminal, following  a crash, not sure if this is related to the cause, or simply occured because the app crashed however:
```
###!!! [Parent][DispatchAsyncMessage] Error: PBackgroundIDBFactory::Msg_PBackgroundIDBFactoryRequestConstructor Value error: message was deserialized, but contained an illegal value
```
- The crashes seem to occur more often when launching from a `.desktop` in the gnome menu, compared to using the CLI. This is just a gut feeling at the moment though, and may be due to chance.

I'll have another poke tomorrow. I'll see if I can get it to crash again with the `-- --jsconsole` flag, and may also try building directly from source to see if that makes any difference.

**Comment by filips123** [comment]
Please update the native program to [v2.13.2](https://github.com/filips123/PWAsForFirefox/releases/tag/v2.13.2), run "patch profiles and runtime" from the extension settings and check if this issue happens again. I've fixed some other issue, but it might also be related to this one.

Note that the extension will say that it's outdated, but this is fine until the new version is also reviewed by Mozilla.

**Comment by nemonein** [comment]
This happens for Google Apps. (Keep, Calendar, etc)
I'm on Archlinux with KDE. firefoxpwa 2.13.2, Progressive Web Apps for Firefox is still 2.13.1.

I've found a workaroud(?) for this.
On Firefox settings, these must be **ON**.
* Open previous windows and tabs
* Show browser tabs and enable using multi-tabbed web apps

But there's one condition.
You have to close FF by Ctrl+W, not Alt-F4. (With mouse click, just click the tab's X button, not the Window's X.)
If you do it with Alt-F4, you will see duplicated tabs.

**Comment by mlncn** [comment]
This is happening with me Harvest (`https://example-businessname.harvestapp.com/`) but not my other 'apps' (Big Blue Button rooms through https://meet.coop mostly).  Can confirm that nemonein's workaround works.

**Comment by wroyca** [comment]
Soft ping @filips123 – one reproducible scenario I've found is to check "Open out-of-scope URLs in a default browser (can break some web apps)" in the Firefox webapp settings and then rebooting. FWIW, I've confirmed with Slack and Discord.

**Comment by franckgaga** [comment]
Same thing on Fedora 42. Can confirm that removing the `%u` in the `.desktop` launcher solve the issue. Maybe add an option to remove the `%u` on launcher files ? It's not a super annoying bug, except if the PWA is configured to start at logon (e.g. in gnome tweaks app). In such case, the PWA does not start at logon

**Comment by filips123** [comment]
So actually, is the problem that web apps close immidiately after opening, or that "relative URL without a base" error when launching from the terminal? Because that error is unrelated, as `%u` is replaced with the URL by desktop environment when launching on a specific URL, or an empty string otherwise. And when it's empty, it doesn't affect launching web apps in any way.

**Comment by AndersKANYE** [comment]
Yes, the web apps close immediately after opening - both when using my DE and through the terminal. I'm on Arch with KDE, using firefoxpwa 2.14.1.
The fix which is mentioned by omitting the `%u` is not changing anything for me.

If it helps at all, it happens when using the Microsoft Teams and Snapchat PWA's.

**Comment by franckgaga** [comment]
It seems to be triggered when I log out while my webapp are still opened. This is what's happening after I re-logged in:

https://github.com/user-attachments/assets/4f1a42a8-bb7e-46c4-8a91-a327b6f120d5

My `Messenger.desktop` file execute this (default settings):
```
/usr/bin/firefoxpwa site launch 01JSJC3YY6JH0M87Y0Q90TPMDQ --protocol %u
```

It may be related to the "restore session" features of firefox.

edit: I get exactly the same issue if I start the webapp with:

```bash
firefoxpwa site launch 01JSJC3YY6JH0M87Y0Q90TPMDQ
```

**Comment by filips123** [comment]
Oh, I can also reproduce it in this way! I still don't know why it happens though, as the console doesn't show any errors.

**Comment by Himura2la** [comment]
I can reproduce this on RHEL 10 @ GNOME 47

<img width="1564" height="948" alt="Image" src="https://github.com/user-attachments/assets/6413c4fe-89ab-4c61-ace0-bbdcb71a39c5" />

**Comment by filips123** [comment]
Has anyone managed to reproduce this without logging out? It's kinda annoying having to relog every time I want to test this.

**Comment by franckgaga** [comment]
No, it's really the first execution after login in. It may seems like a minor issue but it prevent adding a web app as startup application. I personally use it for google chat.

**Comment by nightshadecf** [comment]
it is definitely a bug, whenever I restart cachyos with pwa already opened, the system wont be able to launch app and install launch an empty FireFoxPWA window.
This bug is annoying because it prevents restore all previous apps opened when your system restart.
I believe it is the as the "first-launching failure" mentioned here

**Comment by nroj95** [comment]
same issue here on kubuntu 26.04 + kde wayland with pwasforfirefox 2.19.0.

i found that it happens when a pwa is still open during reboot: after login, the first launch fails, while the second launch works normally. if i close the pwas before rebooting, they usually open fine on the first try.

removing --protocol %u did not fix it for me either.
